Modules/LDAPObject.c

```diff
@@ -268,8 +265,16 @@ attrs_from_List( PyObject *attrlist, char***attrsp, PyObject** seq) {
             "expected *list* of strings, not a string", attrlist);
         goto error;
     } else {
-        *seq = PySequence_Fast(attrlist, "expected list of strings or None");
-        if (*seq == NULL)
+        PyObject *item = NULL;
+        Py_ssize_t i, len, strlen;
+#if PY_MAJOR_VERSION >= 3
```

Comment:
That's wrong, it must be ``#if PY_VERSION_HEX >= 0x03070000``. Python < 3.7 has ``PyAPI_FUNC(char *) PyUnicode_AsUTF8AndSize()``. Only 3.7+ has ``PyAPI_FUNC(const char *) PyUnicode_AsUTF8AndSize()``

Reply:
I thought implicitly converting `char *` to `const char *` is perfectly fine – it just limits what can be done with the string later.
Why is `const char *str = PyUnicode_AsUTF8AndSize(...)` a problem on 3.6-?

Reply:
I discussed the issue with @vstinner. He assured me that const is always fine here. Let's go for your version check.